test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,379 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <test_util.h>
+
+#define MAX_KEYS 100000
+
+static const char *const conn_config =
+  "create,cache_size=100MB,log=(archive=false,enabled=true,file_max=100K)";
+static const char *const table_config = "key_format=S,value_format=S";
+
+static const char *const full_out = "./backup_full";
+static const char *const home_full = "WT_HOME_LOG_FULL";
+static const char *const home_incr = "WT_HOME_LOG_INCR";
+static const char *const home_incr_copy = "WT_HOME_LOG_INCR_COPY";
+static const char *const home_live = "WT_HOME_LOG";
+static const char *const incr_out = "./backup_incr";
+static const char *const uri = "table:logtest";
+
+static const char *wt_tool_path = "./wt";
```

Comment:
I would not expect the `wt` program to be in the same directory as the test. It is okay for this test to know its relative path compared to the `wt` program. It is not really okay to copy the executable into here in order to be able to use `./wt`. It should default to `../../wt` or whatever is the correct relative address. No other program needs to do this so it is unexpected to me.

Reply:
I agree that copying executables is not acceptable.